libpoolprotocols/PoolManager.h

```diff
@@ -48,7 +48,7 @@ namespace dev
 			PoolClient *p_client;
 			Farm &m_farm;
 			MinerType m_minerType;
-			std::chrono::steady_clock::time_point m_submit_time;
+			std::queue<std::chrono::steady_clock::time_point> m_submit_times;
```

Comment:
You can change this to a list instead of a queue, since you don't need the random access supported by the queue class. This would allow the use of clear above.

Reply:
> since you don't need the random access 

Sorry I should learn to read twice.
Yes I do not need random access.

Reply:
Sensible is a pretty subjective term... so is elegant :-)